osu.Game/Rulesets/UI/RulesetInputManager.cs

```diff
@@ -222,14 +231,34 @@ protected override void ReloadMappings(IQueryable<RealmKeyBinding> realmKeyBindi
                 RealmKeyBindingStore.ClearDuplicateBindings(KeyBindings);
             }
         }
+
+        private class ReplayStateReset : IInput
+        {
+            public void Apply(InputState state, IInputStateChangeHandler handler)
+            {
+                if (!(state is RulesetInputManagerInputState<T> inputState))
+                    throw new InvalidOperationException($"{nameof(ReplayState<T>)} should only be applied to a {nameof(RulesetInputManagerInputState<T>)}");
+
+                new MouseButtonInput([], state.Mouse.Buttons).Apply(state, handler);
+                new KeyboardKeyInput([], state.Keyboard.Keys).Apply(state, handler);
+                new TouchInput(Enum.GetValues<TouchSource>().Select(s => new Touch(s, Vector2.Zero)), false).Apply(state, handler);
+                new JoystickButtonInput([], state.Joystick.Buttons).Apply(state, handler);
+                new MidiKeyInput(new MidiState(), state.Midi).Apply(state, handler);
+                new TabletPenButtonInput([], state.Tablet.PenButtons).Apply(state, handler);
+                new TabletAuxiliaryButtonInput([], state.Tablet.AuxiliaryButtons).Apply(state, handler);
```

Comment:
See https://discord.com/channels/188630481301012481/188630652340404224/1258027155208536115 for rationale for this atrocity.

Reply:
I would like to think that this is a workaround for a bug caused by the way replay input handling / `KeyBindingContainer.Trigger{Pressed,Released}` works, and is fine to exist for now. 

Replay input handler should have full control over the ruleset input manager's input state, but currently they only mutate the state of the `KeyBindingContainer`, and that's not good because the state of `KeyBindingContainer` will still be influenced by the input manager's input state, which the `ReplayInputHandler` does not touch.

When user switches from gameplay to autoplay while holding a key, `KeyBindingContainer.TriggerReleased` will get called on the action associated with the key, but nothing happens because KBC still sees the corresponding key in a pressed state and because simultaneous binding mode != all (see [discord](https://discord.com/channels/188630481301012481/188630652340404224/1258035390141567057) for more info).

Either the replay input handler should update the state of the input manager rather than the key binding container, or `TriggerPressed`/`TriggerReleased` should be disassociated with the state of the parenting input manager by some way. None of these methods seem that simple to work with, so I'm fine with performing a manual reset here for now.